### PR TITLE
Gutenboarding: Remove disabled vertical input

### PR DESCRIPTION
**Pull request by tyxla**
#### Changes proposed in this Pull Request

This is a follow-up cleanup to #44983 that disabled the vertical input. ("Vertical" in this context means the industry vertical a site might belong to https://public-api.wordpress.com/wpcom/v2/verticals/  ) The `gutenboarding/show-vertical-input` feature flag has been disabled since, so this PR is cleaning up code that's hidden behind it.

Note: We're leaving the verticals data store in place - should it be removed too?

#### Testing instructions

* Verify the removed code is not in use.
* Verify Calypso builds well and tests pass.
* Start on `/new` and verify you don't see a vertical input.

### CODE REVIEW

**Comment by simison** [comment]
Just noting that we have some designs in process where we might bring this back in 2022, but meanwhile let's indeed remove the code :-)

File: client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx [hunk 1]
```diff
@@ -80,13 +77,9 @@ const SiteTitle: React.FunctionComponent< Props > = ( { onSubmit, inputRef } ) =
...
-
 	// translators: label for site title input in Gutenboarding
 	let inputLabel;
-	if ( showVerticalInput && ! isMobile ) {
-		inputLabel = __( "It's called" );
-	} else if ( isAnchorFmSignup ) {
+	if ( isAnchorFmSignup ) {
 		inputLabel = __( 'My podcast is called' );
 	} else {
 		inputLabel = __( 'My site is called' );
```
Comment (jsnajdr) [inline client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx]: The conditional could be transformed to a ternary: `const inputLabel = ... ? __( ... ) : __( ... )`

**Review by jsnajdr — APPROVED** [review]
> We're leaving the verticals data store in place - should it be removed too?

Yes, while we're removing this, let's remove everything.

File: client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx [hunk 2]
```diff
@@ -1,48 +1,31 @@
...
 	const { __ } = useI18n();
-	const {
-		getSelectedVertical,
-		getSelectedSiteTitle,
-		wasVerticalSkipped,
-		hasSiteTitle,
-	} = useSelect( ( select ) => select( STORE_KEY ) );
+	const { getSelectedVertical, getSelectedSiteTitle, hasSiteTitle } = useSelect( ( select ) =>
+		select( STORE_KEY )
+	);
```
Comment (alshakero) [inline client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx]: Is this still needed?
```suggestion
	const { getSelectedSiteTitle, hasSiteTitle } = useSelect( ( select ) =>
		select( STORE_KEY )
	);
```

File: client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx [hunk 3]
```diff
@@ -1,48 +1,31 @@
...
 
-	const showSiteTitleAndNext = !! (
-		getSelectedVertical() ||
-		hasSiteTitle() ||
-		wasVerticalSkipped()
-	);
-
 	useTrackStep( 'IntentGathering', () => ( {
 		selected_vertical_slug: getSelectedVertical()?.slug,
 		selected_vertical_label: getSelectedVertical()?.label,
```
Comment (alshakero) [inline client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx]: Shouldn't these go as well?

**Review by alshakero — APPROVED** [review]
This looks good, I think there are a few more remnants that will never be removed if kept after this PR, but I imagine it's pretty hard to remove them with certainty that things won't break, given we use the onboarding store in multiple places around checkout. 

1. [This action creator](https://github.com/Automattic/wp-calypso/blob/remove/gutenboarding-vertical-input/client/landing/gutenboarding/stores/onboard/actions.ts#L128-L130). 
2. [These action creators](https://github.com/Automattic/wp-calypso/blob/remove/gutenboarding-vertical-input/client/landing/gutenboarding/stores/onboard/actions.ts#L203-L211).
2. [This reducer](https://github.com/Automattic/wp-calypso/blob/remove/gutenboarding-vertical-input/client/landing/gutenboarding/stores/onboard/reducer.ts#L196-L198).
4. [This reducer](https://github.com/Automattic/wp-calypso/blob/remove/gutenboarding-vertical-input/client/landing/gutenboarding/stores/onboard/actions.ts#L204-L211).
5. [These props in types](https://github.com/Automattic/wp-calypso/blob/remove/gutenboarding-vertical-input/client/landing/gutenboarding/lib/analytics/types.ts#L47-L55). 
4. [The SiteVertical type](https://github.com/Automattic/wp-calypso/blob/remove/gutenboarding-vertical-input/client/landing/gutenboarding/stores/onboard/types.ts#L1).
5. [The vertical persistence settings](https://github.com/Automattic/wp-calypso/blob/remove/gutenboarding-vertical-input/client/landing/gutenboarding/stores/onboard/index.ts#L34-L35).

File: client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx (hunk 2 above)
Comment (tyxla) [reply client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx]: Nope, good find 👍

File: client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx (hunk 3 above)
Comment (tyxla) [reply client/landing/gutenboarding/onboarding-block/acquire-intent/index.tsx]: Good point. I've cleaned those up.

File: client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx (hunk 1 above)
Comment (tyxla) [reply client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx]: Indeed, updated and moved inline.

**Comment by tyxla** [comment]
Thanks for taking a look @jsnajdr and @alshakero! 🙌 

I've cleaned up the verticals from the data store, their unused types, and some unused verticals CSS as well. 

This is ready for another review. Thanks!

**Comment by simison** [comment]
Just for the record I wanted to add visuals how this looked like once we get back to re-doing this next year:

https://user-images.githubusercontent.com/87168/146756680-4b59e94d-7cd1-4e5d-a5de-b04254249331.mov

https://user-images.githubusercontent.com/87168/146756682-f345ae56-7688-4413-9bf3-d7379f9861b5.mov

**Comment by autumnfjeld** [comment]
@Automattic/ganon-bloggers  if anyone has time for a quick review on cleaning up some onboarding code. :)

File: client/landing/gutenboarding/onboarding-block/acquire-intent/style.scss [hunk 4]
```diff
@@ -116,11 +110,6 @@
 	}
 }
 
-.acquire-intent__mobile-vertical-input {
-	display: flex;
-	flex-wrap: wrap;
-}
-
 .acquire-intent__mobile-next-arrow,
 .acquire-intent__mobile-back-arrow {
```
Comment (p-jackson) [inline client/landing/gutenboarding/onboarding-block/acquire-intent/style.scss]: These classes are also no longer used and can be removed.

File: client/landing/gutenboarding/stores/onboard/actions.ts [hunk 5]
```diff
@@ -201,15 +188,6 @@ export const setSiteTitle = ( siteTitle: string ) => ( {
...
-export const setSiteVertical = ( siteVertical: SiteVertical ) => ( {
-	type: 'SET_SITE_VERTICAL' as const,
-	siteVertical,
-} );
-
-export const skipSiteVertical = () => ( {
-	type: 'SKIP_SITE_VERTICAL' as const,
-} );
-
 export const togglePageLayout = ( pageLayout: Template ) => ( {
```
Comment (p-jackson) [inline client/landing/gutenboarding/stores/onboard/actions.ts]: I searched and couldn't find any use of this action. It's sort of outside the scope of this PR, but if we remove this action then we can remove the `Template` type, which would remove the last remaining import from the vertical data store.
I don't know what `togglePageLayout` is for, but given it's something to do with a vertical template, I suspect it's not something we need.

**Review by p-jackson — COMMENTED** [review]
Smoke tested Gutenboarding paying particular attention to features that used to use the vertical data. Everything working as expected.

> We're leaving the verticals data store in place - should it be removed too?

I think we should leave it in place. The verticals data store is in another package with the intention that it's not tied to any particular user experience. We will be re-adding some vertical customisation to one of our onboarding experiences some times this year I think (not this exact feature afaik though), and it could be that the vertical data store is the best way to access it.
We should clean up the last remaining import of it into Gutenboarding though, which I've pointed out below.

File: client/landing/gutenboarding/onboarding-block/acquire-intent/style.scss (hunk 4 above)
Comment (tyxla) [reply client/landing/gutenboarding/onboarding-block/acquire-intent/style.scss]: Good find, removed in https://github.com/Automattic/wp-calypso/pull/59307/commits/7b79b78a1b3ca8098373580597aa662c48d0aaa0.

File: client/landing/gutenboarding/stores/onboard/actions.ts (hunk 5 above)
Comment (tyxla) [reply client/landing/gutenboarding/stores/onboard/actions.ts]: Good catch! Cleaned up in https://github.com/Automattic/wp-calypso/pull/59307/commits/889bba6464d574ff755d44d217aed5a3775a974e.

**Comment by tyxla** [comment]
> Smoke tested Gutenboarding paying particular attention to features that used to use the vertical data. Everything working as expected.

Thanks a bunch for taking a look!

> > We're leaving the verticals data store in place - should it be removed too?
> 
> I think we should leave it in place. 

Fine by me, leaving it intact.

> We should clean up the last remaining import of it into Gutenboarding though, which I've pointed out below.

Addressed by the last commits.

This PR could use another final look! 👀

**Review by p-jackson — APPROVED** [review]
LGTM. Thanks for the tidy up.

**Pull request merged**